conans/client/graph/proxy.py

```diff
@@ -66,7 +66,12 @@ def _get_recipe(self, reference, check_updates, update, remote_name, recorder):
         check_updates = check_updates or update
         # Recipe exists in disk, but no need to check updates
         if not check_updates:
-            status = RECIPE_INCACHE
+            if self._client_cache.installed_as_editable(reference):
```

Comment:
The RECIPE_EDITABLE status should be irrespective of the check_updates, please verify this, it seems wrong.